src/plugins/csvstorage/csvstorage.c

```diff
@@ -0,0 +1,350 @@
+/**
+ * \file
+ *
+ * \brief Source for csvstorage plugin
+ *
+ * \copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+
+#ifndef HAVE_KDBCONFIG
+# include "kdbconfig.h"
+#endif
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <kdberrors.h>
+
+#include "csvstorage.h"
+
+static char *parseLine(char *_line, char delim, unsigned long offset)
+{
+	char *line = (_line + offset);
+	if(*line == '\0')
+		return NULL;
+	char *ptr = strchr(line, delim);
+
+	if(ptr == NULL)
+	{
+		line[strlen(line)-1]='\0';
+	}
+	else
+		*ptr = '\0';
+	return line;
+}
+
+static unsigned long getLineLength(FILE *fp)
+{
+	int startPos = ftell(fp);
+	char c;
+	while((c = fgetc(fp)) && (!feof(fp)))
+	{
+		if(c == '\n')
+			break;
+	}
+	int  endPos = ftell(fp);
+	fseek(fp, startPos, SEEK_SET);
+	if((endPos - startPos) == 0)
+		return 0;
+	else
+		return (endPos - startPos)+1;
+}
+
+static unsigned long getColumnCount(char *lineBuffer, char delim)
+{
+	char *ptr = lineBuffer;
+	unsigned long counter = 0;
+	while(*ptr != '\0')
+	{
+		if(*ptr == delim)
+			++counter;
+		++ptr;
+	}
+	++counter;
+	return counter;
+}
+
+char *itostr(char *buf, unsigned long i, uint8_t len)
+{
+	snprintf(buf, len, "%lu", i);
+	return buf;
+}
+
+static Key *getKeyByOrderNr(KeySet *ks, unsigned long n)
+{
+	Key *cur;
+	char buf[16];
+	ksRewind(ks);
+	while((cur = ksNext(ks)) != NULL)
+	{
+		if(!strcmp(keyString(keyGetMeta(cur, "csvOrder")), itostr(buf, n, sizeof(buf)-1)))
```

Comment:
Please document csvOrder in doc/METADATA.ini, even if its only used internally. And I would prefer `csv/order`.